client/src/components/InputsModal/InputOAuthCredentials.tsx

```diff
@@ -63,46 +52,70 @@ const InputOAuthCredentials: FC<InputOAuthCredentialsProps> = ({
     </Fragment>
   );
 
-  const oAuthField = (field: InputOAuthField) => (
-    <ListItem disabled={requirement.locked} key={field.name}>
-      <TextField
-        disabled={requirement.locked}
-        id={`requirement${index}_${field.name}`}
-        label={field.label || field.name}
-        helperText={requirement.description}
-        value={oAuthCredentials[field.name]}
-        className={styles.inputField}
-        variant="standard"
-        fullWidth
-        onChange={(event) => {
-          const value = event.target.value;
-          inputsMap.set(requirement.name, value);
-          oAuthCredentials[field.name] = value;
-          inputsMap.set(requirement.name, JSON.stringify(oAuthCredentials));
-          setInputsMap(new Map(inputsMap));
-        }}
-        InputLabelProps={{ shrink: true }}
-      />
-    </ListItem>
-  );
+  const oAuthFields: InputOAuthField[] = [
+    { name: 'access_token', label: 'Bearer Token', required: true },
+    {
+      name: 'refresh_token',
+      label: 'Refresh Token (will automatically refresh if available)',
+    },
+    {
+      name: 'token_url',
+      label: 'Token Endpoint',
+      required: showRefreshDetails,
```

Comment:
We should check with @arscan , but I don't think we can make these be required when they're shown.

Reply:
We chatted offline a while ago and said that these should be required fields, let me know if that changes

Reply:
At a minimum, I think requiring `expires_in` is potentially problematic because it isn't required in the original token response: http://hl7.org/fhir/smart-app-launch/1.0.0/#step-3-app-exchanges-authorization-code-for-access-token

Reply:
If credientials aren't required, I wouldn't expect any of the credential fields to be required.  Right now, when I set the demo test to have the creds input be `optional: true`, I still have to put in an access token, which is odd if the credentials are optional. 
<img width="604" alt="Screen Shot 2022-01-21 at 3 45 36 PM" src="https://user-images.githubusercontent.com/412901/150597802-75b8bad0-8e2f-4e45-925e-12e90ed8655f.png">

Reply:
Should be resolved now!